sia/components/hostdb.go

```diff
@@ -20,18 +20,17 @@ const (
 // interfaces.
 
 type HostDB interface {
+	// FlagHost alerts the HostDB that a host is not behaving as expected. The
+	// HostDB may decide to remove the host, or just reduce the weight, or it
+	// may decide to ignore the flagging. If the flagging is ignored, an error
+	// will be returned explaining why.
+	FlagHost(id string) error
```

Comment:
An error probably isn't terribly useful here, unless you expect the flagger to change their behavior depending on whether or not their flag was accepted. I imagine down the road we'll want more sophisticated flagging logic, but I don't think errors are of much use there either.